web/src/app/users/UserDetails.js

```diff
@@ -19,14 +21,14 @@ import Spinner from '../loading/components/Spinner'
 import { GenericError, ObjectNotFound } from '../error-pages'
 import { useConfigValue, useSessionInfo } from '../util/RequireConfig'
 import AppLink from '../util/AppLink'
-import PageActions from '../util/PageActions'
+import UserEditDialog from './UserEditDialog'
 import UserDeleteDialog from './UserDeleteDialog'
-import OtherActions from '../util/OtherActions'
 
 const userQuery = gql`
   query userInfo($id: ID!) {
     user(id: $id) {
       id
+      role
```

Comment:
I think we can safely remove `role` from the queries here if we use our `useSessionInfo` hook (more info below on this)

Reply:
As mentioned in an earlier comment, there needs to be a way to propogate the `role` of the user whose details page is being viewed. The `isAdmin` from `useSessionInfo` is if the logged in user is an admin or not, vs the role of the user for details page. That's why I thought querying for `role` and passing it onto `UserEditDialog` would be the best option.